add eq function to NodeSource

Gasoonjia opened the pull request:
Summary: add eq function to NodeSouce by comparing their dict representation.

Test Plan:
ci

Rollback Plan:

Differential Revision: D78200762




cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

yushangdi reviewed (approved):
there's a lint failures, otherwise LGTM

Gasoonjia committed:
add eq function to NodeSource (#158170)

Summary:

add eq function to NodeSouce by comparing their dict representation.

Test Plan:
ci

Rollback Plan:

Reviewed By: yushangdi, ezyang

Differential Revision: D78200762
